rpm_s3_mirror/mirror.py

```diff
@@ -144,6 +145,24 @@ def snapshot(self, snapshot_id):
                     self._try_remove_snapshots(snapshot_id=snapshot_id)
                     raise Exception("Failed to snapshot repositories") from e
 
+    def list_snapshots(self):
+        snapshots = defaultdict(dict)
+        for repository in self.repositories:
+            repo_path = urlparse(repository.base_url).path
+            prefix = join(repo_path, "snapshots")
+            results = self.s3.list(prefix=prefix[1:])
+            for result in results:
+                key, last_modified = result["Key"], result["LastModified"]
+                # We write the repomd.xml file last so this time
+                # is when the snapshot is considered complete.
+                if key.endswith("repomd.xml"):
+                    match = SNAPSHOT_REGEX.search(key)
+                    if match:
```

Comment:
I guess could just do the regex first and combine the ifs and it won't be much slower since the regex is compiled, but that's a minor detail, not requesting changes.